c10/util/safe_numerics.h

```diff
@@ -61,7 +66,7 @@ bool safe_multiplies_u64(It first, It last, uint64_t* out) {
   uint64_t prod = 1;
   bool overflow = false;
   for (; first != last; ++first) {
-    overflow |= c10::mul_overflows(prod, *first, &prod);
+    overflow |= c10::mul_overflows(prod, static_cast<uint64_t>(*first), &prod);
```

Comment:
requiring these casts looks like a bc break. I would recommend resolving it by providing an overload with the exact old signature:
```
C10_ALWAYS_INLINE bool mul_overflows(uint64_t a, uint64_t b, uint64_t* out) {
    return mul_overflows<uint64_t>(a, b, out);
}
```

(non-templates are preferred over templates during overload resolution, and you're explicitly specifying template parameters in the implementation so there's no ambiguity -- https://godbolt.org/z/9MYb3Kdhh)